ImperatorToCk3/Source/Imperator/Characters/Character.h

```diff
@@ -0,0 +1,70 @@
+#ifndef IMPERATOR_CHARACTER_H
+#define IMPERATOR_CHARACTER_H
+#include "Date.h"
+#include "Parser.h"
+
+namespace ImperatorWorld
+{
+class Family;
+
+typedef struct
+{
+	int martial = 0;
+	int finesse = 0;
+	int charisma = 0;
+	int zeal = 0;
+} AttributesStruct;
+
+class Character: commonItems::parser
+{
+  public:
+	Character(std::istream& theStream, int chrID);
+
+	[[nodiscard]] const std::string& getCulture() const;
+	[[nodiscard]] const std::string& getReligion() const { return religion; }
+	[[nodiscard]] const auto& getName() const { return name; }
+	[[nodiscard]] const auto& getBirthDate() const { return birthDate; }
+	[[nodiscard]] const auto& getDeathDate() const { return deathDate; }
+	[[nodiscard]] const auto& getSpouses() const { return spouses; }
+	[[nodiscard]] const auto& getChildren() const { return children; }
+	[[nodiscard]] const auto& getMother() const { return mother; }
+	[[nodiscard]] const auto& getFather() const { return father; }
+	[[nodiscard]] const auto& getFamily() const { return family; }
+	[[nodiscard]] const auto& getTraits() const { return traits; }
+	[[nodiscard]] const auto& getAttributes() const { return attributes; }
+
+	[[nodiscard]] auto isFemale() const { return female; }
+	[[nodiscard]] auto getWealth() const { return wealth; }
+	[[nodiscard]] auto getID() const { return charID; }
+
+	void setFamily(std::shared_ptr<Family> theFamily) { family.second = std::move(theFamily); }
```

Comment:
You shouldn't be needing to move from a shared_ptr.

Reply:
I took this line from CK2ToEU4: `void setDynasty(std::shared_ptr<Dynasty> theDynasty) { dynasty.second = std::move(theDynasty); }`
I will look for a way to do it differently, since I don't really understand std::move();

Reply:
Nono, leave it. Shared_ptr was already copied when given to the function as argument so it's internal counter was increased. moving it prevents additional copying.

With shared pointers this is desired behavior - we don't reference to them, they exist to be copied and shared like this.

Reply:
That's right. We had a whole conversation about it too.